File: Algorithms/283. Move Zeroes/golang/solution.space-sub-optimal.go
func moveZeroes(nums []int) {
	var n, numZeroes int = len(nums), 0
	for i := 0; i < n; i++ {
		if nums[i] == 0 {
			numZeroes++
		}
	}
	var ans []int
	for _, num := range nums {
		if num != 0 {
			ans = append(ans, num)
		}
	}
	for numZeroes > 0 {
		ans = append(ans, 0)
		numZeroes--
	}
	for i := 0; i < n; i++ {
		nums[i] = ans[i]
	}
}

File: Algorithms/283. Move Zeroes/python/solution.space-sub-optimal.py
class Solution:
    def moveZeroes(self, nums: list[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)

        # Count the zeroes
        num_zeroes = 0
        for i in range(n):
            num_zeroes += nums[i] == 0

        # Make all the non-zero elements retain their original order.
        ans = list()
        for num in nums:
            if num != 0:
                ans.append(num)

        # Move all zeroes to the end
        while num_zeroes > 0:
            ans.append(0)
            num_zeroes -= 1

        # Combine the result
        for i in range(n):
            nums[i] = ans[i]
